static/js/containers/LoginButton.js

```diff
@@ -34,12 +35,12 @@ class LoginButton extends React.Component {
 
 LoginButton.propTypes = {
   dispatch: React.PropTypes.func.isRequired,
-  authentication: React.PropTypes.object.isRequired
+  userProfile: React.PropTypes.object.isRequired
 };
 
 const mapStateToProps = (state) => {
   return {
-    authentication: state.authentication
+    userProfile: state.userProfile
```

Comment:
could we change this to `profile: state.userProfile.profile`? if we only need things on the profile object (`profile.preferred_name`) this might be cleaner.